### ISSUE TITLE
Fix signal handler

### ISSUE THREAD

**Issue by elfring** [Netflix/dynomite#135]
The functions "[_log](https://github.com/Netflix/dynomite/blob/2c6b25a3e7ac85bc942c886211ac2b72d9c9e3fa/src/dyn_log.h#L86 "Check loga()")" and "[exit](http://pubs.opengroup.org/onlinepubs/9699919799/functions/exit.html "Terminate a process")" do not belong to the list of [async-signal-safe functions](https://www.securecoding.cert.org/confluence/display/c/SIG30-C.+Call+only+asynchronous-safe+functions+within+signal+handlers "Call only asynchronous-safe functions within signal handlers.").
I guess that a different program design will be needed for your function "[signal_handler](https://github.com/Netflix/dynomite/blob/2c6b25a3e7ac85bc942c886211ac2b72d9c9e3fa/src/dyn_signal.c#L71)".

**Comment by timiblossom** [comment]
Yes, it is not a good way.  Let's keep this issue open for anyone want to work on this.
